src/component/requests.js

```diff
@@ -15,32 +17,69 @@ export default class Requests extends React.Component {
     });
   }
 
+  componentDidMount() {
+    const requests = React.findDOMNode(this);
+    requests.addEventListener('scroll', () => {
+      const fromIndex = Math.ceil(requests.scrollTop / requestElementHeight) - 15;
+      this._setFromIndex(fromIndex < 0 ? 0 : fromIndex);
+    });
+  }
+
+  componentDidUpdate() {
+    const previousFilter = this.filter;
+    this.filter = this.props.requestData.filter;
+    const requests = React.findDOMNode(this);
```

Comment:
Actually I would call `React.findDomNode` only within the condition. I don't see it being used outside of the condition. Leads to less function calls, maybe even less DOM operations and increases readability.